### PR TITLE
Check that certain consul config values are present and valid

### PR DESCRIPTION
**Pull request by roberteckert**
https://jira.rbx.com/browse/ICSD-386

### CODE REVIEW

**Comment by roberteckert** [comment]
This list of configuration values is generated from the various policy_roles _consul_agent.rb files. I only am only checking the values that are configured via DynamicAttributes.

**Review by shivdudhani — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
 
+## 9003.1.25
+* Designate certain config values as mandatory
+  - acl_master_token
+  - acl_token
+  - acl_agent_token
+  - advertise_addr
+  - advertise_addr_wan
+  - bind_addr
+  - encrypt
```
Comment (sherrman) [inline CHANGELOG.md:11]: What about `license_content`? From 1.10.7-ent and later, if they do not have `license_content` the agent will fail to start.

**Review by sherrman — APPROVED** [review]
Thanks for this PR, it's great!

Left a note for an optional change. The change for that is more complex than the existing ones since it doesn't fully apply to older versions.

File: CHANGELOG.md (hunk 1 above)
Comment (roberteckert) [reply CHANGELOG.md:11]: This is partially covered here: https://github.com/Roblox/consul-cookbook/blob/master/libraries/consul_config.rb#L304.

I can postpend something like `&& Gem::Version.new(node["consul"]["version"]) >= Gem::Version.new("1.10.7-ent")` to the line, but I would like to decompose that into its own change.

**Comment by roberteckert** [comment]
I, after some struggle, added the test suites that worked on my mac (w/ parallels). I only really verified with Ubuntu 20.04, but I think that these results are adequately representative of what we will see.

I also consolidated the kitchen configuration to make use of the preexisting default-config label.

**Pull request merged**